classes/migration/upgrade/v3_4_0/I9627_AddUsageStatsTemporaryTablesIndexes.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I9627_AddUsageStatsTemporaryTablesIndexes.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I9627_AddUsageStatsTemporaryTablesIndexes
+ *
+ * @brief Add an index to temporary usage stats tables to fix/improve the removeDoubleClicks and compileUniqueClicks query.
+ */
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\migration\Migration;
+
+class I9627_AddUsageStatsTemporaryTablesIndexes extends Migration
+{
+    /**
+     * Run the migration.
+     */
+    public function up(): void
+    {
+        $sm = Schema::getConnection()->getDoctrineSchemaManager();
+        Schema::table('usage_stats_total_temporary_records', function (Blueprint $table) use ($sm) {
+            $indexesFound = $sm->listTableIndexes('usage_stats_total_temporary_records');
+            if (!array_key_exists('ust_load_id_context_id_ip', $indexesFound)) {
+                switch (DB::getDriverName()) {
+                    case 'mysql':
```

Comment:
I think this way is safer `if (DB::connection() instanceof PostgresConnection) {` (i.e. using an identifier instead of a string)